api/controllers/SdtdServer/get-players-datatable.js

```diff
@@ -1,46 +1,39 @@
 // If you don't know wtf is going on here
 // https://datatables.net/manual/server-side
 
+const sevenDays = require('machinepack-7daystodiewebapi');
+
 async function getPlayersDataTable(req, res) {
+
   const whereObj = {
     server: req.body.serverId
   };
   const totalPlayers = await Player.count(whereObj);
 
-  const queryObj = {
-    where: whereObj,
-    select: req.body.columns.map(c => c.data),
-    skip: parseInt(req.body.start),
-    limit: parseInt(req.body.length),
-  };
-
-  if (!_.isEmpty(req.body.search.value)) {
-    whereObj.or = [{
-      name: {
-        contains: req.body.search.value
-      }
-    },
-    {
-      steamId: {
-        contains: req.body.search.value
-      }
-    },
-    {
-      ip: {
-        contains: req.body.search.value
-      }
+  if (!!req.body.searchText) {
+    whereObj.or = [
+      { name: { contains: req.body.searchText } },
+      { steamId: { contains: req.body.searchText } },
+      { ip: { contains: req.body.searchText } }
+    ]
+    if (!isNaN(req.body.searchText)) {
+      whereObj.or.push({ entityId: req.body.searchText });
     }
-    ];
-
-
   }
-  if (req.body.order.length) {
-    const colNumToSort = parseInt(req.body.order[0].column);
-    const sortObj = {};
-    sortObj[req.body.columns[colNumToSort].data] = req.body.order[0].dir;
-    queryObj.sort = [sortObj];
+  if (req.body.onlyOnline === 'true') {
+    const server = await SdtdServer.findOne(req.body.serverId);
+    const playerList = await getPlayerList(server);
```

Comment:
Good this is gated behind the onlyOnline parameter. This can get quite heavy otherwise with the infinite scrolling 😄